src/shadowbox/server/server_access_key.ts

```diff
@@ -212,8 +219,7 @@ export class ServerAccessKeyRepository implements AccessKeyRepository {
     for (const accessKey of this.accessKeys) {
       const usageBytes = bytesTransferredById[accessKey.id] || 0;
       const wasOverDataLimit = accessKey.isOverDataLimit;
-      accessKey.isOverDataLimit =
-          this.accessKeyDataLimit ? usageBytes > this.accessKeyDataLimit.bytes : false;
+      accessKey.isOverDataLimit = usageBytes > (accessKey.dataLimit || this.defaultDataLimit)?.bytes || false;
```

Comment:
Can you extract the limit to a separate variable and keep the ternary operator to avoid `|| false`?

Reply:
Done

Reply:
The ternary has a small bug that it skips the check and returns false if the data limit is 0, since 0 is falsy.

I extracted to a separate variable for readability.